pkg/apis/kops/componentconfig.go

```diff
@@ -366,28 +366,28 @@ type KubeAPIServerConfig struct {
 	// OIDCUsernameClaim is the OpenID claim to use as the user name.
 	// Note that claims other than the default ('sub') is not guaranteed to be
 	// unique and immutable.
-	OIDCUsernameClaim *string `json:"oidcUsernameClaim,omitempty" flag:"oidc-username-claim"`
+	OIDCUsernameClaim *string `json:"-" flag:"oidc-username-claim"`
```

Comment:
Is this just because we don't want to be serializing from the unversioned form, or is it something deeper?

Reply:
We only want to take values from the field in `authentication.oidc`. The field exists here for the `flag` annotation: so we can pass it to kube-apiserver. Longer term, we should probably have different struct types for the cluster API and the thing that passes data to kube-apiserver.